### ISSUE TITLE
Not tracking afk when using barrier

### ISSUE THREAD

**Issue by ashirviskas** [ActivityWatch/activitywatch#704]
- [x] I am on the [latest](https://github.com/ActivityWatch/activitywatch/releases/latest) ActivityWatch version.
 - [x] I have searched the issues of this repo and believe that this is not a duplicate.

- **OS name and version**: Arch
- **ActivityWatch version**: v0.11.0

## Describe the bug
When using [barrier](https://github.com/debauchee/barrier) (allows for multiple computer to share a single mouse/keyboard), the afk tracking is broken on the host computer, as it (most likely) still registers mouse and keyboard movements in another computer.

## To Reproduce
1. Install [barrier](https://github.com/debauchee/barrier)
2. Connect 2 PCs
3. Do something using host PC mouse/keyboard in the client PC
4. Activity is tracked on host PC

## Expected behavior
1. Activity isn't tracked on host PC and marked as afk as soon as the cursor leaves the display

**Comment by ErikBjare** [comment]
This seems impossible to do something about without integrating barrier somehow (which I'm not personally interested in). I assume something similar applies to Synergy and other input-sharing software.

Closing.

**Comment by ashirviskas** [comment]
@ErikBjare Gotcha. I still haven't looked at the code, but is it possible to check where the cursor is in AW and if it's not found, mark it as afk? I might as well just fork it and do it myself if it doesn't require a major rewrite

**Comment by ErikBjare** [comment]
@ashirviskas Yeah maybe, you could try modifying this function to check if cursor is OK a screen border or something and then not could activity on/outside the border as AFK (or however the mouse gets treated with barrier): https://github.com/ActivityWatch/aw-watcher-afk/blob/7ce0d6b389ed87acdf7ad0876b9372c769b63f66/aw_watcher_afk/unix.py#L20

**Comment by ashirviskas** [comment]
@ErikBjare Thanks! 

And if I was about to write a separate custom watcher, could I override the afk watcher using api? Because I see that I can read buckets, write to them and delete them, but is it possible to modify them? Or would this pseudocode work?

```
bucket_id = "{}_{}".format("afk", client.client_hostname)

def barrier_callback(event_type):
        if event_type == 'cursor_left_display':
            event_data = {"label": "afk"}
        else:
            event_data = {"label": "not-afk"}

        now = datetime.now(timezone.utc)
        event = Event(timestamp=now, data=event_data)
        inserted_event = client.insert_event(bucket_id, event)
            

```

**Comment by ErikBjare** [comment]
I'm not sure I understand the question, but that code example doesn't make sense to me.

I'd just modify the aw-watcher-afk watcher to do that check directly, before any event is sent to the server, and then run that instead of the default aw-watcher-afk watcher.

**Comment by ashirviskas** [comment]
Sorry if I wasn't being clear, let me rephrase and improve it:

Can I write a separate watcher, which would track whether the current host is active in Barrier (it would be nice if I could avoid replicating the current aw-watcher-afk fuctionality) and use that together with unmodified AW, by only modifying configs?

That would allow me to easily share it with the community.

**Comment by ErikBjare** [comment]
I wouldn't recommend that approach. It would make things a lot more complicated as you'd have to modify the queries used for analysis (makes it a lot harder to share/reuse, as queries are hard-coded in the web UI).

**Comment by ashirviskas** [comment]
@ErikBjare  That's too bad... Any other way you'd recommend? Or could I just make the default aw-watcher-afk `poll_time=999999999` (would somewhere in the range of 10 years) and copy its functionality to my custom `barrier-watcher`, including the bucket name?

**Comment by ErikBjare** [comment]
As I said before, I'd just fork aw-watcher-afk and add the check there. 

Not sure why you would modify poll_time, I don't see how that would help (it would effectively just disable the watcher).

**Comment by ashirviskas** [comment]
My bad, I thought you couldn't disable `aw-watcher-afk` as it wasn't in the settings, but I missed the fact that you can easily disable modules from the tray icon options... :facepalm: Thanks, will do just that.

**Comment by ErikBjare** [comment]
@ashirviskas It can also be disabled through the config file `~/.config/activitywatch/aw-qt/aw-qt.toml`:

```
[aw-qt]
autostart_modules = ["aw-server", "aw-watcher-window"]
#
#[aw-qt-testing]
#autostart_modules = ["aw-server-rust", "aw-watcher-afk", "aw-watcher-window"]
```

You could also put a build of the fork in your PATH under a new name `aw-watcher-afk-barrier`, and then add a `"aw-watcher-afk-barrier"` entry to `autostart_modules`, and that way it would automatically be started with aw-qt.

**Comment by ashirviskas** [comment]
@ErikBjare Thank you! It's exactly what I was just looking into, saved me some time.
